assets/scripts/app/ScrollIndicators.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { registerKeypress } from './keypress'
+
+class ScrollIndicators extends React.Component {
+  static propTypes = {
+    posLeft: PropTypes.number.isRequired,
+    posRight: PropTypes.number.isRequired,
+    scrollStreet: PropTypes.func.isRequired,
+    scrollTop: PropTypes.number.isRequired
+  }
+
+  shouldComponentUpdate (nextProps) {
+    if (this.props.posLeft !== nextProps.posLeft &&
+        this.props.posRight !== nextProps.posRight) {
+      this.refs.left_indicator.innerHTML = Array(nextProps.posLeft + 1).join('‹')
+      this.refs.right_indicator.innerHTML = Array(nextProps.posRight + 1).join('›')
```

Comment:
I think we can do inside the `render()` function itself, so we don't need to set `refs` for each element, and we can also avoid using `innerHTML`. (It doesn't matter as much here, because this doesn't get filled with user input, but avoiding it is still a good idea, especially since `render()` was designed to take care of this.)